prboom2/src/v_video.c

```diff
@@ -1285,16 +1285,17 @@ void SetRatio(int width, int height)
   ratio_scale *= 3;
   ReduceFraction(&ratio_multiplier, &ratio_scale);
 
+  {
+    float ratio_quotient = (float)ratio_multiplier/ratio_scale;
+    float ratio_percentage = (ratio_quotient - 1) * 100.0;
+    psprite_offset = MAX(0, (int)(ratio_percentage*FRACUNIT));
```

Comment:
This math only needs to be done when `tallscreen` is true, right? Let's put it inside the tallscreen block and leave the original else block in place.

Reply:
Correct. My initial commit had the math inside the block but I wanted to get a little fancy and use MAX to make the else branch unnecessary.

I'll revert my later commits.